packages/wordcount/src/index.js

```diff
@@ -111,18 +129,19 @@ function matchCharacters( text, regex, settings ) {
  */
 
 export function count( text, type, userSettings ) {
-	if ( '' === text ) {
-		return 0;
+	const settings = loadSettings( type, userSettings );
+	let matchRegExp;
+	if ( settings.type === 'words' ) {
```

Comment:
Depending on your feelings about `switch` statements, the function logic looks like a good candidate:

```js
switch ( settings.type ) {
	case 'words':
		// ...

	case 'characters_including_spaces':
		// ...

	// ...
}
```

Reply:
I tried the `switch` and the `if`s looked more aesthetically pleasing to me in this specific case. However, it just occurred to me that we can skip the `break`s, because we will have a `return` in each case and that actually tips my aesthetic scales towards `switch`. Thank you! Done in 0ac3aff.